Workday/canvas-kit
# fix(pagination): Fixes onPageChange regression

## Summary

Fixes: #1745

![category](https://img.shields.io/badge/release_category-Components-blue)

---

## Overview

This PR fixes an issue where `onPageChange` was firing on any model event, even if the page was the current page.

## Where Should the Reviewer Start?

`modules/react/pagination/lib/Pagination/usePaginationModel.ts`

## Testing Manually

- Visit the Basic Pagination story: `?path=/story/components-navigation-pagination-react--basic`
- Open the browser JS console
- Update the current page by using the page buttons or the controls
- See the console log the current page when onPageChange is called
- It should no longer be called when you click on the current page button or when the current page is unchanged

## Screenshots

![a demo in storybook of the fix](http://g.recordit.co/PYO6GhqxJ2.gif)

## Thank You Gif

![John Oliver takes in the moment as thousands of pages rain from the air](https://media.giphy.com/media/cNfdCiR00n8kCKeuhJ/giphy.gif)
